packages/common/src/api/tan-query/useCollectionByParams.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react'
+
+import { useDispatch } from 'react-redux'
+
+import { ID } from '~/models/Identifiers'
+import {
+  collectionPageActions,
+  collectionPageLineupActions
+} from '~/store/pages'
+
+import { Config } from './types'
+import { useCollection } from './useCollection'
+import { useCollectionByPermalink } from './useCollectionByPermalink'
+
+type CollectionParams =
+  | { collectionId?: ID }
+  | { handle?: string; slug?: string; collectionType?: 'album' | 'playlist' }
+  | { permalink?: string }
+
+// feature-tan-query TODO: smart collections
+
+/**
+ * Hook that returns collection data given either a collection ID, handle+slug+type, or permalink.
+ * Internally uses useCollection and useCollectionByPermalink for consistent behavior.
+ * @param params The collection params - either {collectionId}, {handle, slug, type}, or {permalink}
+ * @returns The collection data or null if not found
+ */
+export const useCollectionByParams = (
+  params: CollectionParams,
+  options?: Config
+) => {
+  const dispatch = useDispatch()
+
+  let permalink: string | null | undefined = null
+  if ('permalink' in params) {
+    permalink = params.permalink
+  } else if ('handle' in params) {
+    permalink = `/${params.handle}/${params.collectionType}/${params.slug}`
+  }
+
+  const collectionId = 'collectionId' in params ? params.collectionId : null
+
+  const idQuery = useCollection(collectionId, options)
+  const permalinkQuery = useCollectionByPermalink(permalink, options)
+
+  const query = collectionId ? idQuery : permalinkQuery
+
+  const { isSuccess } = query
+  const collectionIdResult = query.data?.playlist_id
+
+  useEffect(() => {
+    if (isSuccess && collectionIdResult) {
+      dispatch(
+        collectionPageActions.fetchCollectionSucceeded(collectionIdResult)
+      )
+
+      dispatch(
+        collectionPageLineupActions.fetchLineupMetadatas(
```

Comment:
this again is starting to feel very collectionPage specific. Another point in favor of storing the code in an appropriate location. No action for this PR though 👍

Reply:
It is collection page/screen specific because it's taking params from the url, I just wanted to share this in mobile and web